### ISSUE TITLE
add serial number to disk object

### ISSUE THREAD

**Issue by jeroenmaelbrancke** [openvstorage/framework#1093]
Can you add the disk serial number to the Disk object?

```
{
    "is_ssd": false, 
    "vendor": null, 
    "name": "sdd", 
    "storagerouter_guid": "a47b712a-1077-4860-ae3c-6d50d350348b", 
    "state": "OK", 
    "path": "/dev/disk/by-id/wwn-0x50014ee2acbf4781", 
    "model": "WDC_WD10EACS-00D6B1", 
    "guid": "3a711355-541d-40ee-9f1a-b9fce754e7f9", 
    "size": 1000204886016
}
```

If a disk is missing in the kernel, we would be able to retrieve the disk information via the model.

for example:
Disk /dev/sdd is missing in the kernel

```
Oct 27 11:14:08 ovs02 kernel: [16573330.608615] ata4.00: status: { DRDY }
Oct 27 11:14:08 ovs02 kernel: [16573330.634924] ata4: hard resetting link
Oct 27 11:14:08 ovs02 kernel: [16573330.953753] ata4: SATA link down (SStatus 0 SControl 300)
Oct 27 11:14:13 ovs02 kernel: [16573335.952559] ata4: hard resetting link
Oct 27 11:14:14 ovs02 kernel: [16573336.272495] ata4: SATA link down (SStatus 0 SControl 300)
Oct 27 11:14:14 ovs02 kernel: [16573336.272512] ata4: limiting SATA link speed to 1.5 Gbps
Oct 27 11:14:19 ovs02 kernel: [16573341.271329] ata4: hard resetting link
Oct 27 11:14:19 ovs02 kernel: [16573341.591320] ata4: SATA link down (SStatus 0 SControl 310)
Oct 27 11:14:19 ovs02 kernel: [16573341.591336] ata4.00: disabled
Oct 27 11:14:19 ovs02 kernel: [16573341.591348] blk_update_request: I/O error, dev sdd, sector 977613683
Oct 27 11:14:19 ovs02 kernel: [16573341.617736] ata4: EH complete
Oct 27 11:14:19 ovs02 kernel: [16573341.617744] XFS (sdd1): metadata I/O error: block 0x3a452373 ("xlog_iodone") error 5 numblks 64
Oct 27 11:14:19 ovs02 kernel: [16573341.644693] XFS (sdd1): xfs_do_force_shutdown(0x2) called from line 1177 of file /build/linux-lts-wily-edDYjE/linux-lts-wily-4.2.0/fs/xfs/xfs_log.c.  Return address = 0xffffffffc0656080
Oct 27 11:14:19 ovs02 kernel: [16573341.644701] XFS (sdd1): Log I/O Error Detected.  Shutting down filesystem
Oct 27 11:14:19 ovs02 kernel: [16573341.671136] XFS (sdd1): Please umount the filesystem and rectify the problem(s)
```

We can filter, via the Disk object, the disk and retrieve the serial number of the disk.

**Comment by wimpers** [comment]
@khenderick how much work would this be?

**Comment by khenderick** [comment]
Most likely it won't be reliable, since serial numbers tend to be stored all over the place. But I don't think it will be much work.

**Comment by wimpers** [comment]
@kinvaris and rest of OPS:
please advise on  how to reliable retrieve the serial nr.

**Comment by jeroenmaelbrancke** [comment]
Smartctl will cover the most of the physical hardware (nvme, ata, ssd,....)

```
root@ovs-esxi-host1:~# smartctl -d scsi -i /dev/nvme0n1
smartctl 6.2 2013-07-26 r3841 [x86_64-linux-4.2.0-22-generic] (local build)
Copyright (C) 2002-13, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF INFORMATION SECTION ===
Vendor:               NVMe
Product:              INTEL SSDPEDME40
Revision:             0110
User Capacity:        400,088,457,216 bytes [400 GB]
Logical block size:   512 bytes
Rotation Rate:        Solid State Device
Logical Unit id:      8086INTEL SSDPEDME400G4                     1000CVMD4456001S400AGN  
Serial number:        CVMD4456001S400AGN
Device type:          disk
Local Time is:        Thu Oct 27 14:42:20 2016 CEST
SMART support is:     Unavailable - device lacks SMART capability.
```

**Comment by dejonghb** [comment]
newer lsblk (16.04) might also be nice:

```
$ lsblk -J -S -o NAME,SERIAL
{
   "blockdevices": [
      {"name": "sda", "serial": "TW0MGNHV5508556A0449"}
   ]
}
```

**Comment by wimpers** [comment]
@khenderick can we use Bert's suggestion?

**Comment by khenderick** [comment]
@wimpers, no idea, I guess so. In any case, it will be a "best effort" solution, not a works-for-all-possible-disks solution.

**Comment by wimpers** [comment]
@jeroenmaelbrancke  would this be ok for you?

**Comment by jeroenmaelbrancke** [comment]
Sure

**Comment by khenderick** [comment]
Fixed by #1201, packaged in openvstorage-2.7.6-rev.4348.457b7ba

**Comment by kinvaris** [comment]
PASSED: Check on one of our CI environments
```
In [3]: print DiskList.get_disks()[0]
{
    "size": 10737418240, 
    "is_ssd": false, 
    "name": "sdd", 
    "storagerouter_guid": "e8c9cffd-bd96-4cb8-8c98-64b59bd804dc", 
    "state": "OK", 
    "model": "QEMU HARDDISK", 
    "guid": "04c679a9-487a-4274-bf00-da7268253bba", 
    "serial": "7423e15a-979d-11e6-b",                                       <-------------------------
    "aliases": [
        "/dev/disk/by-path/pci-0000:00:06.0-ata-4", 
        "/dev/disk/by-id/ata-QEMU_HARDDISK_7423e15a-979d-11e6-b"
    ]
}
```
